### PR TITLE
stm32: add blues wireless swan

### PR DESCRIPTION
**Pull request by kenbell**
I've verified:
* Timer setup (blink frequency)
* I2C using BME280
* Serial IO (UART)

### CODE REVIEW

**Comment by deadprogram** [comment]
Check this out @rayozzie :smile_cat:

**Comment by kenbell** [comment]
Updated to use stm32 dfu mode by default...
```tinygo flash -target swan <path>```  << use DFU
```tinygo flash -target swan -programmer openocd <path>``` << use stlink

File: src/machine/machine_stm32l4.go [hunk 1]
```diff
@@ -117,6 +117,25 @@ const (
 	PE15 = portE + 15
 )
 
+// IRQs are defined here as they vary in the SVDs, but do have consistent mapping
+// to Timer Interrupts.
+const (
+	irq_TIM1_BRK_TIM15     = 24
```
Comment (deadprogram) [inline src/machine/machine_stm32l4.go:123]: Perhaps you want to put these into `device/stm32` similar to how https://github.com/tinygo-org/tinygo/blob/release/src/device/sam/atsamd51x-bitfields.go is?

Comment (kenbell) [reply src/machine/machine_stm32l4.go:123]: How strongly do you feel about it?

I kinda feel icky about putting hand-crafted code in there.  Long-term better to work with rust team on getting the IRQ constants aligned.  I've made these non-exported so they don't become an implicit API for apps.

Comment (deadprogram) [reply src/machine/machine_stm32l4.go:123]: I like your long-term solution much better!

**Comment by deadprogram** [comment]
Thanks for another great contribution @kenbell now merging.

**Comment by rayozzie** [comment]
Indeed, my deep appreciation!

**Pull request merged**